apps/site/pages/en/learn/modules/package-configuration.md

```diff
@@ -0,0 +1,563 @@
+| ESM: source code uses `import`    | CJS & ESM: consumers either `require()` or `import` your package | [ESM source and both CJS & ESM distribution](#esm-source-and-both-cjs-amp-esm-distribution)<br />[ESM source and distribution](#esm-source-and-distribution) |
+
+### CJS source and distribution
+
+This the "Rum & Coke" of packages: pretty difficult to mess up. You _technially_ may not need _any_ package configuration beyond [`"name"`](https://nodejs.org/api/packages.html#name). But the less arcane, the better: Essentially just declare the package’s exports via the `"exports"` field/field-set.
+
+**Working example**: [cjs-with-cjs-distro](https://github.com/JakobJingleheimer/nodejs-module-config-examples/tree/main/packages/cjs/cjs-distro)
+
+```json displayName="Minimal package.json"
+{
+  "name": "cjs-source-and-distribution"
+  // "main": "./index.js"
+}
+```
+
+```json displayName="Advanced (verbose) package.json"
+{
+  "name": "cjs-source-and-distribution",
+  "type": "commonjs", // current default, but may change
+  "engines": { "node": ">=12.22.7" },
+  "exports": {
+    ".": "./dist/index.js",
+    "./package.json": "./package.json"
+  }
+}
+```
+
+Note that `packageJson.exports["."] = filepath` is shorthand for `packageJson.exports["."].default = filepath`
+
+### CJS source and only ESM distribution
+
+The "Gin & Tonic" of packages: This takes a small bit of finesse but is also pretty straight-forward.
+
+**Working example**: [cjs-with-esm-distro](https://github.com/JakobJingleheimer/nodejs-module-config-examples/tree/main/packages/cjs/esm-distro)
+
+```json displayName="Minimal package.json"
+{
+  "name": "cjs-source-with-esm-distribution",
+  "main": "./dist/index.mjs"
+}
+```
+
+```json displayName="Advanced (verbose) package.json"
+{
+  "name": "cjs-source-with-esm-distribution",
+  "type": "commonjs", // current default, but may change
+  "engines": { "node": ">=12.22.7" },
+  "exports": {
+    ".": "./dist/index.mjs",
+    "./package.json": "./package.json"
+  }
+}
+```
+
+The [`.mjs`](https://nodejs.org/api/esm.html#enabling) file extension is a trump-card: it will override **any** other configuration and the file will be treated as ESM. Using this file extension is necessary because `packageJson.exports.import` does **NOT** signify that the file is ESM (contrary to common, if not universal, misperception), only that it is the file to be used when the package is imported (ESM _can_ import CJS. See [Gotchas](#gotchas) below).
+
+### CJS source and both CJS & ESM distribution
+
+You have a few options:
+
+#### Attach named exports directly onto `exports`
+
+The "French 75" of packages: Classic but takes some sophistication and finesse.
+
+Pros:
+
+- Smaller package weight
+- Easy and simple (probably least effort if you don't mind keeping to a minor syntax stipulation)
+- Precludes the Dual-Package Hazard
```

Comment:
The article previously linked to it in the module docs, but that section has been removed. I was going to ask you if you'd consider returning that specific piece back to the API docs or a Learn article (perhaps even this article?). I think it being in that repo makes it too far removed for anyone to find it.

Reply:
I think the dual package hazard can be explained in this post directly, since most examples about dual shipping are related to it. That information certainly feels more useful compared to some of the opening paragraphs...